rotkehlchen/chain/evm/decoding/balancer/utils.py

```diff
@@ -0,0 +1,232 @@
+                encounter=TokenEncounterInfo(
+                    description=f'Querying {CPT_BALANCER_V1} balances',
+                    should_notify=False,
+                ),
+                underlying_tokens=[
+                    UnderlyingToken(
+                        address=get_or_create_evm_token(
+                            userdb=inquirer.database,
+                            chain_id=inquirer.chain_id,
+                            protocol=CPT_BALANCER_V1,
+                            name=token['name'].strip(),
+                            symbol=token['symbol'].strip(),
+                            evm_address=string_to_evm_address(to_checksum_address(token['address'])),
+                            encounter=TokenEncounterInfo(
+                                description=f'Querying {CPT_BALANCER_V1} balances',
+                                should_notify=False,
+                            ),
+                        ).evm_address,
+                        token_kind=EvmTokenKind.ERC20,
+                        weight=FVal(token['weight']),
+                    )
+                    for token in pool['displayTokens']
+                ],
+            )
+        except (KeyError, ValueError, TypeError, DeserializationError) as e:
+            log.error(
+                f'Failed to process balancer v1 pool on {inquirer.chain_id} due to {e!s}. '
+                f'Pool: {pool}. Skipping...',
+            )
+            continue
+
+        if pool_token.protocol != CPT_BALANCER_V1:
+            log.debug(f'Updating protocol for balancer asset {pool_token}')
+            GlobalDBHandler.set_token_protocol_if_missing(
+                token=pool_token,
+                new_protocol=CPT_BALANCER_V1,
+            )
+
+    with GlobalDBHandler().conn.write_ctx() as write_cursor:
+        globaldb_set_unique_cache_value_at_ts(
+            write_cursor=write_cursor,
+            key_parts=cache_key_parts,
+            value=str(latest_pools_count),
+            timestamp=now,
+        )
+
+
+def query_balancer_api(query: str, variables: dict[str, Any]) -> dict[str, Any]:
+    """Make a request to Balancer API with standardized error handling.
+    Raises:
+    - RemoteError: If the request fails or response is invalid
+    """
+    try:
+        response = requests.post(
+            url=BALANCER_API_URL,
+            json={
+                'query': query,
+                'variables': variables,
+            },
+            headers={'User-Agent': 'rotkehlchen'},
+            timeout=CachedSettings().get_timeout_tuple(),
+        )
+        if response.status_code != 200:
+            raise RemoteError(f'Balancer API request failed with status {response.status_code}')
+
+        data = response.json()
+        if 'errors' in data:
+            raise RemoteError(f"Balancer API query failed: {data['errors']}")
+
+        return data['data']
+    except (requests.exceptions.RequestException, KeyError) as e:
```

Comment:
it can't throw `JSONDecodeError` as its handled in the method.
```py
    def json(self, **kwargs):
        r"""Returns the json-encoded content of a response, if any.

        :param \*\*kwargs: Optional arguments that ``json.loads`` takes.
        :raises requests.exceptions.JSONDecodeError: If the response body does not
            contain valid json.
        """

        if not self.encoding and self.content and len(self.content) > 3:
            # No encoding set. JSON RFC 4627 section 3 states we should expect
            # UTF-8, -16 or -32. Detect which one to use; If the detection or
            # decoding fails, fall back to `self.text` (using charset_normalizer to make
            # a best guess).
            encoding = guess_json_utf(self.content)
            if encoding is not None:
                try:
                    return complexjson.loads(self.content.decode(encoding), **kwargs)
                except UnicodeDecodeError:
                    # Wrong UTF codec detected; usually because it's not UTF-8
                    # but some other 8-bit codec.  This is an RFC violation,
                    # and the server didn't bother to tell us what codec *was*
                    # used.
                    pass
                except JSONDecodeError as e:
                    raise RequestsJSONDecodeError(e.msg, e.doc, e.pos)

        try:
            return complexjson.loads(self.text, **kwargs)
        except JSONDecodeError as e:
            # Catch JSON-related errors and raise as requests.JSONDecodeError
            # This aliases json.JSONDecodeError and simplejson.JSONDecodeError
            raise RequestsJSONDecodeError(e.msg, e.doc, e.pos)
```

Reply:
per the discussion in discord, i'm thinking of writing a pylint plugin. what do you think -- an overkill?

Reply:
> per the discussion in discord, i'm thinking of writing a pylint plugin. what do you think -- an overkill?

These comments are not a place for discussion. Move to disord.